pkg/controllers/common/constants.go

```diff
@@ -116,6 +116,13 @@ const (
 	TemplateGeneratorMergePatchAnnotation = FederateControllerPrefix + "template-generator-merge-patch"
 
 	LatestReplicasetDigestsAnnotation = DefaultPrefix + "latest-replicaset-digests"
+
+	// MigrationInfoAnnotation contains custom migration information from users on the source object.
+	MigrationInfoAnnotation = DefaultPrefix + "migration-info"
+	// LatestMigrationInfoAnnotation contains latest custom migration information from users on the source object.
+	LatestMigrationInfoAnnotation = DefaultPrefix + "latest-migration-info"
+	// CustomMigrationInfoAnnotation contains custom migration information from users on the federated object.
+	CustomMigrationInfoAnnotation = DefaultPrefix + "custom-migration-info"
```

Comment:
CustomMigrationInfoAnnotation is an internal annotation. Use InternalPrefix instead.